models/remoteworkitem.go

```diff
@@ -0,0 +1,77 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
+	GetWorkItemKeyMap()
+}
+
+// GithubRemoteWorkItem is a derivative of RemoteWorkItem
+type GithubRemoteWorkItem struct {
```

Comment:
> I wouldn't want a common data structure to be touched

The structure is the same, you would only be adding values to the map.

As for now and some time, this will only be statically defined. If you want, hide the 'static map' behind;

```go

type RemoteWorkItemRepository struct {
  mappings map[string]WorkItemMap
}

func (repo *RemoteWorkItemRepository) Get(provider string, type WorkItemType) WorkItemMap {
  return repo.mappings[provider];
}

func NewRemoteWorkItemRepository() RemoteWorkItemRepository {
  return RemoteWorkItemRepository{
    mappings: map[string]WorkItemMap{"github": WorkItemMap{....}},
  }
}

```